packages/config/types.d.ts

```diff
@@ -0,0 +1,125 @@
+declare module "@onflow/fcl" {
+  export type Environment = "local" | "canarynet" | "testnet" | "mainnet";
+
+  type AccountProofData = {
+    // e.g. "Awesome App (v0.0)" - A human readable string to identify your application during signing
+    appIdentifier: string;
+    // e.g. "75f8587e5bd5f9dcc9909d0dae1f0ac5814458b2ae129620502cb936fde7120a" - minimum 32-byte random nonce as hex string
+    nonce: string;
+  };
+  type AccountProofDataResolver = () => Promise<AccountProofData | null>;
+
+  /**
+   * @see {@link https://docs.onflow.org/fcl/reference/api/#common-configuration-keys}
+   * @see {@link https://docs.onflow.org/fcl/reference/configure-fcl/}
+   */
+  export interface ConfigurationOptions {
+    /**
+     * API URL for the Flow Blockchain Access Node you want to be communicating
+     * with. See all available access node endpoints
+     * [here]{@link https://docs.onflow.org/access-api/#flow-access-node-endpoints}.
+     * @example "https://rest-testnet.onflow.org"
+     */
+    "accessNode.api": string;
+    /**
+     * Your applications title, can be requested by wallets and other services.
+     * @example "Cryptokitties"
+     */
+    "app.detail.title"?: string;
+    /**
+    * Url for your applications icon, can be requested by wallets and other
+    * services.
+    * @example "https://fcl-discovery.onflow.org/images/blocto.png"
+    */
+    "app.detail.icon"?: string;
+    /**
+     * Points FCL at the Wallet or Wallet Discovery mechanism.
+     * @deprecated Use `discovery.wallet` instead.
+     */
+    "challenge.handshake"?: string;
+    /**
+     * Endpoint for alternative configurable Wallet Discovery mechanism.
+     * Read more on [discovery]{@link https://docs.onflow.org/fcl/reference/api/#discovery}
+     * @example "https://fcl-discovery.onflow.org/api/testnet/authn"
+     */
+    "discovery.authn.endpoint"?: string;
+    /**
+     * Points FCL at the Wallet or Wallet Discovery mechanism.
+     * @example "https://fcl-discovery.onflow.org/testnet/authn"
+     */
+    "discovery.wallet": string;
+    /**
+     * Used in conjunction with stored interactions.
+     * @deprecated
+     */
+    env?: Environment;
+    /**
+     * Specifies fallback compute limit if not provided in transaction.
+     * @example 100
+     */
+    "fcl.limit"?: number;
+    /**
+     * Used in conjunction with stored interactions and provides 
+     * FCLCryptoContract address for `testnet` and `mainnet`.
+     */
+    "flow.network"?: Environment;
+    // TODO: these options are not documented; are there more?
+    "discovery.wallet.method"?: string;
+    "fcl.accountProof.resolver"?: AccountProofDataResolver;
+    "service.OpenID.scopes"?: string
```

Comment:
I have found these keys in the different doc examples. Are there more that we want to include here for better DevEx?

Reply:
`fcl.accountProof.resolver` use is documented [here](https://docs.onflow.org/fcl/reference/proving-authentication/)

others include:

- "fcl.storage": accepts a storage option object
```
const LOCAL_STORAGE = {
  can: !isServerSide(),
  get: async key => JSON.parse(localStorage.getItem(key)),
  put: async (key, value) => localStorage.setItem(key, JSON.stringify(value)),
}
```
- "debug.accounts" : boolean
- "logger.level" : 1 | 2 | 3 | 4 | 5
https://github.com/onflow/fcl-js/tree/master/packages/util-logger